src/ts/batch/MultiSend.ts

```diff
@@ -0,0 +1,49 @@
+import { MinimalTx } from "../forwarders/forwarder";
+import { MultiSendFactory } from "../../typedContracts/MultiSendFactory";
+import { MULTI_SEND_ADDRESS } from "../../deployment/addresses";
+import { Signer } from "ethers";
+
+/**
+ * Batch a list of meta-transactions before it hits the forwarder.
+ * Useful for deploying proxy contract and then sending meta-tx
+ * in a single Ethereum Transaction.
+ */
+export class MultiSender {
+  /**
+   * Given a list of minimal transactions, it'll prepare a single
+   * minimal transaction that is sent via the MultiSend contract.
+   * @param signer Signer's wallet
+   * @param batch List of minimal transactions
+   * @returns A minimal transaction for the MultiSend contract
+   */
+  public async batch(signer: Signer, batch: MinimalTx[]): Promise<MinimalTx> {
```

Comment:
You can actually get hold of the interface without a signer here - Typechain could be doing better here

```
import { abi } from "../../typedContracts/MultiSend.json"
import { Interface } from "ethers/utils"
import { MultiSend } from "../../typedContracts/MultiSend";

const multiSendFace = new ethers.utils.Interface(abi) as MultiSend["interface"];
const encodedTransactions = multiSendFace.functions.batch.encode([
      to,
      data,
      revertIfFail,
    ]);
```

Reply:
fixed,. so the function is no longer async too.